src/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramTitleFormField/DeclaredProgramTitleFormField.test.ts

```diff
@@ -0,0 +1,182 @@
+import type { AddDeclaredProgramForm, DeclaredProgramFormData } from '@/features/student/declaredPrograms/types/forms.types'
+import DeclaredProgramTitleFormField from '@/features/student/declaredPrograms/components/interactions/formFields/DeclaredProgramTitleFormField/DeclaredProgramTitleFormField.vue'
+import { DECLARED_PROGRAM_TITLE_MAX_LENGTH } from '@/features/student/declaredPrograms/config'
+import { BddTest } from '@avenirs-esr/avenirs-dsav/test-utils'
+import { useForm } from '@tanstack/vue-form'
+import { mount, type VueWrapper } from '@vue/test-utils'
+import { beforeEach, expect, vi } from 'vitest'
+
+const DeclaredProgramTitleInputStub = defineComponent({
+  name: 'DeclaredProgramTitleInput',
+  props: {
+    modelValue: String,
+    errorMessage: String
+  },
+  emits: ['update:modelValue', 'blur'],
+  template: `
+    <div data-testid="declared-program-title-input-stub">
+      <input
+        :value="modelValue"
+        @input="$emit('update:modelValue', $event.target.value)"
+        @blur="$emit('blur')"
+      />
+    </div>
+  `
+})
+
+const TestWrapper = defineComponent({
+  components: {
+    DeclaredProgramTitleFormField
+  },
+  setup () {
+    const form = useForm({
+      defaultValues: {
+        title: ''
+      } as DeclaredProgramFormData,
+      validators: {
+        onSubmit ({ value }) {
+          if (!value.title || value.title.trim() === '') {
+            return {
+              fields: {
+                title: 'Le titre est requis'
+              }
+            }
+          }
+          return { fields: { title: undefined } }
+        }
+      }
+    }) as unknown as AddDeclaredProgramForm
+
+    return { form }
+  },
+  template: `
+    <form @submit.prevent="form.handleSubmit">
+      <DeclaredProgramTitleFormField :form="form" />
+    </form>
+  `
+})
+
+BddTest().given('a declared program title form field', () => {
+  let wrapper: VueWrapper<InstanceType<typeof TestWrapper>>
+
+  const stubs = {
+    DeclaredProgramTitleInput: DeclaredProgramTitleInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the title input component', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramTitleInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have empty initial value', () => {
+      const input = wrapper.findComponent({ name: 'DeclaredProgramTitleInput' })
+      expect(input.props('modelValue')).toBe('')
+    })
+  })
+
+  BddTest().and('the user enters a title', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(TestWrapper, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters a title', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété, valable pour tous les and